src/main/cpp/src/parse_uri.cu

```diff
@@ -44,8 +44,8 @@ constexpr bool is_valid_character(char ch, bool alphanum_only)
     if (ch >= 'A' && ch <= 'Z') return true;               // A-Z
     if (ch >= 'a' && ch <= 'z') return true;               // a-z
   } else {
-    if (ch >= '!' && ch <= ';' && ch != '"') return true;  // 0-9 and !#%&'()*+,-./
-    if (ch >= '=' && ch <= 'Z' && ch != '>') return true;  // A-Z and =?@
+    if (ch >= '!' && ch <= ':' && ch != '"') return true;  // 0-9 and !#%&'()*+,-./:
+    if (ch >= '=' && ch <= ']' && ch != '>') return true;  // A-Z and =?@[]
```

Comment:
[question] not about this change in particular, but I am curious if this code will be faster if you make it a branch-less return of OR-ed lines
```C++
  return (ch >= '!' && ch <= ':' && ch != '"') ||
            (ch >= '=' && ch <= ']' && ch != '>') ||
...
         ;
```

Reply:
godbolt produces the same PTX for both versions. Probably noise differences.